cardano-db-sync/src/Cardano/DbSync/Cache/Types.hs

```diff
@@ -45,6 +46,8 @@ import Ouroboros.Consensus.Cardano.Block (StandardCrypto)
 
 type StakePoolCache = Map PoolKeyHash DB.PoolHashId
 
+type CacheTx = Map ByteString DB.TxId
```

Comment:
Keeping many small ByteString long term in memory is ill-advised, since they can cause memory fragmentation. That's why all ledger types are backed by ShortByteString and we should do the same. Also since ledger has all types in ShortByteString, we avoid the ShortByteString -> ByteString transition when we query something in the cache.

I just realised we use ByteString for `cStakeRawHashes` which should also change.

Reply:
Just to note the existing cache is:
```
newtype TxCache = TxCache {txIdCache :: Map ByteString DB.TxId}
```

So you're saying we need to use `ShortByteString` but try not to convert them back and forth often. But the code I changed uses `ByteString` as it's checking against `TxHash` in `queryTxId`. 

This means I'd have to convert `ByteString -> ShortByteString` every time I add a new value to the `Map`.

I'm unsure how I can use `ShortByteString` in the cache without doing the conversion?

Reply:
The initial type for the hash of a tx is `TxId StandardCrypto` defined at
https://github.com/IntersectMBO/cardano-ledger/blob/f4119c9cd27e69943dcf91d22d0f9905be32fa75/libs/cardano-ledger-core/src/Cardano/Ledger/TxIn.hs#L56

Note that we use functions like `safeHashToByteString` and `unTxHash` to turn it into a ByteString before using  `queryTxId`. This transition may happen even before the creation of the Generic.Tx type currently.

I propose changing the Cache type to
```
newtype TxCache = TxCache {txIdCache :: Map (Ledger.TxId StandardCrypto) DB.TxId}
```

(I initially mentioned `ShortByteString` thinking that it's used to define `SafeHash`, but we don't need to go that deep. `TxId` is good enough)

Reply:
Another note to avoid confusion. This definition
```
unTxHash :: TxId era -> ByteString
```

is a bit misleading. It should be
```
unTxHash :: TxId c -> ByteString
```